app/models/vm_or_template.rb

```diff
@@ -51,7 +51,7 @@ class VmOrTemplate < ApplicationRecord
   validates_presence_of     :name, :location
   validates_inclusion_of    :vendor, :in => VENDOR_TYPES.values
 
-  has_one                   :miq_server, :foreign_key => :vm_id
+  has_one                   :miq_server, :foreign_key => :vm_id, :inverse_of => :vm
```

Comment:
The reporting code only detects relationships from the parent (`VmOrTemplate`). I  remember having trouble with this when working with Foreman.

So if we move it down a layer, this may have consequences in the views.

I dunno, maybe @Fryguy already fixed this?